puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tfmROR52.groovy

```diff
@@ -0,0 +1,67 @@
+def packages_to_build
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '15'))
+        disableConcurrentBuilds()
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+
+                deleteDir()
```

Comment:
Define "clean the git repo" ?

Reply:
https://github.com/theforeman/foreman-infra/blob/9948b6a1b7488351b706852c5eaae933ea1bace7/puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy#L62-L67

Reply:
I suppose we could, however, if there are lingering artifacts not in git that won't cover cleaning those out?

Reply:
That's exactly what it's cleaning out in tarballsRelease. There we had some extra tarballs remaining from the previous run and adding that fixed the issue. It's equivalent to `git clean` (IIRC that's what it calls under the hood).

Reply:
So turning it around, what's the advantage to git clean vs. straight deleteDir to cleanup everything?

Reply:
it does a `git clean -fdx`, which wipes everything that is either unknown or changed, so that should be sufficient

https://github.com/jenkinsci/git-client-plugin/blob/027d2a0b2f8726d2f08fb747e219e497b06e856c/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L804-L806

given this would save time (the clone is not thrown away, just updated cleanly), I'd +1 that change

Reply:
A git clone can take a long time on some repositories. Cleaning + updating might be a lot faster.

Reply:
If you've added `CleanCheckout` then `deleteDir()` should no longer be needed.